output/sarif.go

```diff
@@ -0,0 +1,496 @@
+		// Remove "data." prefix and convert dots to dashes
+		query = strings.TrimPrefix(query, "data.")
+		query = strings.ReplaceAll(query, ".", "-")
+		return fmt.Sprintf("%s-%s", rType.ruleIDPrefix, query)
+	}
+
+	// Use description if available
+	if desc, ok := result.Metadata["description"].(string); ok {
+		return fmt.Sprintf("%s/%s", rType.ruleIDPrefix, strings.ToLower(strings.ReplaceAll(desc, " ", "-")))
+	}
+
+	// Fallback to base ID if no identifying information is available
+	return rType.ruleIDPrefix
+}
+
+// createRule creates a new SARIF rule from a result
+func createRule(result Result, rType resultType, namespace string) sarifRule {
+	ruleID := getRuleID(result, rType, namespace)
+
+	rule := sarifRule{
+		ID: ruleID,
+		ShortDescription: sarifMessage{
+			Text: result.Message,
+		},
+		Properties: make(map[string]interface{}),
+	}
+
+	// Add policy metadata to rule properties
+	if pkg, ok := result.Metadata["package"].(string); ok {
+		rule.Properties["package"] = pkg
+	}
+	if ruleName, ok := result.Metadata["rule"].(string); ok {
+		rule.Properties["rule"] = ruleName
+	}
+	if query, ok := result.Metadata["query"].(string); ok {
+		rule.Properties["query"] = query
+	}
+	rule.Properties["namespace"] = namespace
+
+	// Add additional rule metadata if available
+	if desc, ok := result.Metadata["description"].(string); ok {
+		rule.FullDescription = &sarifMessage{Text: desc}
+	}
+	if url, ok := result.Metadata["url"].(string); ok {
+		rule.HelpURI = url
+	}
+	if help, ok := result.Metadata["help"].(string); ok {
+		rule.Help = &sarifMessage{Text: help}
+	}
+
+	// Add any remaining metadata to properties
+	for k, v := range result.Metadata {
+		switch k {
+		case "package", "rule", "description", "url", "help", "query":
+			// Skip already processed fields
+			continue
+		default:
+			rule.Properties[k] = v
+		}
+	}
+
+	return rule
+}
+
+// createLocation creates a new SARIF location from a file path
+func createLocation(filePath string) sarifLocation {
+	return sarifLocation{
+		PhysicalLocation: sarifPhysicalLocation{
+			ArtifactLocation: sarifArtifactLocation{
+				URI: filepath.ToSlash(filePath),
+			},
+		},
+	}
+}
+
+// createProperties creates a new properties map with namespace information
+func createProperties(metadata map[string]interface{}, namespace string) map[string]interface{} {
+	properties := make(map[string]interface{})
+	for k, v := range metadata {
+		switch k {
+		case "package", "rule", "description", "url", "help":
+			// Skip rule-level metadata
+			continue
+		case "query", "traces", "outputs":
+			// Include query-related information
+			properties[k] = v
+		default:
+			properties[k] = v
+		}
+	}
+
+	// Always include namespace
+	properties["namespace"] = namespace
+
+	return properties
+}
+
+// processResults processes a slice of results and adds them to the SARIF run
+func processResults(run *sarifRun, results []Result, rType resultType, fileName, namespace string, ruleMap map[string]bool) error {
```

Comment:
Please update this to have an explicit return rather than having an input parameter also be used as an output.